pkg/frontend/querymiddleware/split_by_instant_interval_test.go

```diff
@@ -0,0 +1,239 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package querymiddleware
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/user"
+
+	"github.com/grafana/mimir/pkg/util"
+)
+
+func TestQuerySplittingCorrectness(t *testing.T) {
+	var (
+		numSeries          = 1000
+		numStaleSeries     = 100
+		numHistograms      = 1000
+		numStaleHistograms = 100
+		histogramBuckets   = []float64{1.0, 2.0, 4.0, 10.0, 100.0, math.Inf(1)}
+	)
+
+	tests := map[string]struct {
+		query string
+
+		expectedSplitQueries int
```

Comment:
Looks like not checked. Add metrics and check it. See `pkg/frontend/querymiddleware/querysharding_test.go` for reference.

Reply:
Done in commit https://github.com/grafana/mimir/pull/2469/commits/c76603b21e2118edf264e0dd290168298f58b9a9